core/src/lib.rs

```diff
@@ -6,3 +10,58 @@ mod http;
 pub mod release;
 pub mod storage;
 pub mod tracker;
+
+/// The different games supported by the launcher
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum Game {
+    Jumpy,
+    Punchy,
+}
+
+#[derive(thiserror::Error, Debug)]
+#[error("Invalid game ID")]
+pub struct InvalidGameId;
+
+impl FromStr for Game {
+    type Err = InvalidGameId;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "jumpy" => Ok(Game::Jumpy),
+            "punchy" => Ok(Game::Punchy),
+            _ => Err(InvalidGameId),
+        }
+    }
+}
+
+impl Game {
+    pub fn id(&self) -> &'static str {
+        match self {
+            Game::Jumpy => "jumpy",
+            Game::Punchy => "punchy",
+        }
+    }
+
+    pub fn name(&self) -> &'static str {
+        match self {
+            Game::Jumpy => "Jumpy",
+            Game::Punchy => "Punchy",
+        }
+    }
+
+    pub fn binary_name(&self) -> String {
+        let id = self.id();
+
+        #[cfg(target_os = "windows")]
+        let ext = ".exe";
+        #[cfg(not(target_os = "windows"))]
+        let ext = "";
+
+        format!("{id}{ext}")
+    }
```

Comment:
Not a big deal but can be rewritten as:

```rs
    pub fn binary_name(&self) -> String {
        let id = self.id();
	if cfg!(target_os = "windows") {
            format!("{id}.exe")
        } else {
            id.to_string()
        }
    }
```

I'm okay with both. I leave it up to you.

Reply:
Yeah, yours is better.

I learned while working on WGPU that using `cfg!()` is better sometimes because compiling on one platform will still catch type checker errors for the other platform, but `#[cfg()]` checks will not at all check the other platform code for correctness.